packages/mds-db/client.ts

```diff
@@ -49,33 +50,37 @@ async function setupClient(useWriteable: boolean): Promise<MDSPostgresClient> {
   }
 }
 
+const readOnlyClientLock = new AwaitLock()
 export async function getReadOnlyClient(): Promise<MDSPostgresClient> {
-  if (readOnlyCachedClient && readOnlyCachedClient.connected) {
-    return readOnlyCachedClient
-  }
-
+  await readOnlyClientLock.acquireAsync()
```

Comment:
couldn't `acquireAsync()` throw an error also?

Reply:
Unlikely. Acquiring the lock outside of the `try` is the recommended usage pattern in the README. `mds-repository` has been using this same approach/pattern from inception.

Reply:
sure, I mean it's probably fine, I'm just pointing out that acquiring a lock can/should have a timeout less than `+INFINITY`.